app/controllers/api/v4/converts_controller.rb

```diff
@@ -14,7 +14,7 @@ def create
     end
 
     old_extension = Run.program(@run.program).file_extension
-    new_extension = Run.program(params[:format]).file_extension
+    new_extension = Run.program(params[:format]).try(:file_extension) || params[:format]
```

Comment:
ah right, cool. can we make that more clear in the code, i prefer clarity over flexibility